packages/grafana-ui/src/components/Modal/Modal.tsx

```diff
@@ -7,6 +7,7 @@ import { getModalStyles } from './getModalStyles';
 import { ModalHeader } from './ModalHeader';
 import { IconButton } from '../IconButton/IconButton';
 import { HorizontalGroup } from '../Layout/Layout';
+import FocusTrap from 'focus-trap-react';
```

Comment:
This is a thin wrapper around `focus-trap`. Looking at open issues for that library two caught my eye that might cause us problems:

https://github.com/focus-trap/focus-trap/issues/375 (focus stuck on elements with positive tabIndex)
https://github.com/focus-trap/focus-trap/issues/383 (can break `clipboard.js`. Unless I'm mistaken we use it in [ClipboardButton.tsx](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx#L2))

Might be wise to confirm if these issues affect us before merging.

Reply:
@jackw good catches! I believe for Azure neither of these issues should affect us, but it seems a bit aggressive to force all modals to use this when we haven't really done thorough testing yet. What would you think about making this opt-in for now.